portal/api-authentication-grpc.go

```diff
@@ -0,0 +1,211 @@
+// This file is part of MinIO Kubernetes Cloud
+// Copyright (c) 2019 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package portal
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"encoding/base64"
+
+	pq "github.com/lib/pq"
+	cluster "github.com/minio/m3/cluster"
+	pb "github.com/minio/m3/portal/stubs"
+	uuid "github.com/satori/go.uuid"
+)
+
+type User struct {
+	Tenant   string
+	IsAdmin  bool
+	Password string
+	UUID     string
+}
+
+// Login handles the Login request by receiving the user credentials
+// and returning a hashed token.
+func (s *server) Login(ctx context.Context, in *pb.LoginRequest) (*pb.LoginResponse, error) {
+	log.Printf("Calling Login")
+	// Create Credentials
+	// TODO: validate credentials: username->email, tenant->shortname?
+	var res pb.LoginResponse
+	tenantName := in.GetCompany()
+	email := in.GetEmail()
+	pwd := in.GetPassword()
+
+	// Search for the tenant on the database
+	tenant, err := getTenant(tenantName)
+	if err != nil {
+		err = errors.New("Tenant not found")
+		res.Error = err.Error()
+		return &res, err
+	}
+
+	// Password validation
+	user, ok := getUser(tenant.Name, email)
```

Comment:
this validation should be on the DB

Reply:
Updated by lookin on the db both user email and password and returning ~"wrong credentials" error when failing doing the query.